classes/migration/upgrade/PKPv3_3_0UpgradeMigration.inc.php

```diff
@@ -72,43 +72,39 @@ public function up() {
 
 		// pkp/pkp-lib#6093 Don't allow nulls (previously an upgrade workaround)
 		Capsule::schema()->table('user_settings', function (Blueprint $table) {
-			$table->bigInteger('assoc_type')->default(0)->change();
-			$table->bigInteger('assoc_id')->default(0)->change();
+			$table->bigInteger('assoc_type')->default(0)->nullable(false)->change();
+			$table->bigInteger('assoc_id')->default(0)->nullable(false)->change();
```

Comment:
When we start defining `FOREIGN KEY` relationships properly (https://github.com/pkp/pkp-lib/issues/6093) we won't be able to use `0` IDs as de-facto `NULL` values as they'll cause DB errors. So I'd recommend changing any case where a `0` ID is used so that it's updated to `NULL`, and the schema/migration code is adapted to allow `NULL` entries. That's definitely the case with `user_settings`.

Reply:
(I've commented on the cases I could spot; the ones I haven't mentioned should be fine.)